### PR TITLE
ON-31854 # Added extIdGen and personalisation to form class

### PR DESCRIPTION
**Pull request by jdawg093**

### CODE REVIEW

File: OneBlink.SDK/models/Form.cs [hunk 1]
```diff
@@ -34,7 +34,8 @@ public Form()
             string cancelAction = "BACK",
             string cancelRedirectUrl = null,
             FormServerValidation serverValidation = null,
-            FormExternalIdGeneration externalIdGeneration = null,
+            FormExternalIdGeneration externalIdGenerationOnSubmit = null,
+            FormExternalIdGeneration personalisation = null,
```
Comment (mymattcarroll) [inline OneBlink.SDK/models/Form.cs:38]:
```suggestion
            FormServerValidation personalisation = null,

```

File: OneBlink.SDK/models/Form.cs [hunk 2]
```diff
@@ -245,6 +247,14 @@ public FormExternalIdGeneration externalIdGeneration
         {
             get; set;
         }
+        public FormExternalIdGeneration externalIdGenerationOnSubmit
+        {
+            get; set;
+        }
+        public FormExternalIdGeneration personalisation
```
Comment (mymattcarroll) [inline OneBlink.SDK/models/Form.cs:254]:
```suggestion
        public FormServerValidation personalisation

```

File: docs/models/Form.md [hunk 3]
```diff
@@ -4,29 +4,31 @@
...
+| `cancelRedirectUrl`            | No       | `string`                    | URL to be redirected too, only applies if `cancelAction` is "URL"                             | `null`                            |
+| `serverValidation`             | No       | `FormServerValidation`      | Optional configuration for form submission validation                                         | `null`                            |
+| `externalIdGeneration`         | No       | `FormExternalIdGeneration`  | [DEPRECATED]: Optional configuration for generating externalId on form load                   | `null`                            |
+| `externalIdGenerationOnSubmit` | No       | `FormExternalIdGeneration`  | Optional configuration for generating externalId after serverValidation but before submission | `null`                            |
+| `personalisation`              | No       | `FormExternalIdGeneration`  | Optional configuration for prefilling elements or generating new elements on form load        | `null`                            |
```
Comment (jdawg093) [inline docs/models/Form.md:30]:
```suggestion
| `personalisation`              | No       | `FormServerValidation`  | Optional configuration for prefilling elements or generating new elements on form load        | `null`                            |
```

File: OneBlink.SDK/models/Form.cs [hunk 4]
```diff
@@ -34,7 +34,9 @@ public Form()
             string cancelAction = "BACK",
             string cancelRedirectUrl = null,
             FormServerValidation serverValidation = null,
-            FormExternalIdGeneration externalIdGeneration = null,
+            FormExternalIdGeneration externalIdGenerationOnSubmit = null,
+            FormServerValidation personalisation = null,
```
Comment (mymattcarroll) [inline OneBlink.SDK/models/Form.cs:38]: I think we should make a `FormPersonalisation` class for this. This code looks a bit weird.

File: OneBlink.SDK/models/FormPersonalisation.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
+namespace OneBlink.SDK.Model
+{
+    public class FormPersonalisation
+    {
+        public string type
+        {
+            get; set;
+        }
+        public EndpointConfiguration Configuration
```
Comment (mymattcarroll) [inline OneBlink.SDK/models/FormPersonalisation.cs:9]:
```suggestion
        public EndpointConfiguration configuration
```

**Review by mymattcarroll — APPROVED** [review]

**Pull request merged**